From: Jiang Xin <worldhello.net@gmail.com>
Subject: [PATCH] GitHub alike site publish method

If branch html-pub pushed to the server, and it contains
a file 'LOCATION', then checkout branch html-pub to
LOCATION.

Signed-off-by: Jiang Xin <worldhello.net@gmail.com>

---
 hooks/common/post-update |   34 ++++++++++++++++++++++++++++++++++
 1 个文件被修改，插入 34 行(+)

diff --git a/hooks/common/post-update b/hooks/common/post-update
new file mode 100755
index 0000000..bdfcf40
--- /dev/null
+++ b/hooks/common/post-update
@@ -0,0 +1,34 @@
+#!/bin/sh
+#
+# Publish contents in html-pub branche to deply directory,
+# just like gh-pages branch as to GitHub.
+#
+# To enable this hook, rename this file to "post-update".
+
+# Do git-update-server-info for git dumb protocol.
+dumb=$(git config --bool hooks.update-server-info || echo "false")
+if [ "$dumb" = "true" ]; then
+    git update-server-info
+fi
+
+LOCATION=.LOCATION
+DEPLOY=/var/lib/gitolite/deploy
+
+while [ $# -gt 0 ]; do
+    case "${1#refs/heads/}" in
+    html-pub)
+        target=$(git show $1:$LOCATION 2>/dev/null || echo "$GL_REPO")
+        if [ -n "$target" ]; then
+            target=${target#/}
+            target=$DEPLOY/$target
+            mkdir -p "$target"
+            echo >&2 "checkout to $target..."
+            git --work-tree="$target" reset --hard $1
+        else
+            echo >&2 "File '$LOCATION' not found in branch '${1#refs/heads/}'."
+        fi
+        break
+        ;;
+    esac
+    shift
+done
-- 
tg: (f73e3d7..) t/post-update (depends on: upstream)
